minio/minio
# optimize Listen bucket notification implementation

## Description
optimize Listen bucket notification implementation

## Motivation and Context
this commit avoids lots of tiny allocations, repeated
the channel creates which are performed when filtering 
the incoming events, unescaping a key just for matching.

also, remove deprecated code which is not needed
anymore, avoids unexpected data structure transformations
from the map to slice.

## How to test this PR?
No functionality should change but should be able to
observe higher performance with notification targets
and listen bucket notification for `mc mirror --watch` 

## Types of changes

## Checklist:
